### PR TITLE
Allow to configure subscription spec config

### PR DESCRIPTION
**Pull request by machi1990**
## Description

- [allow to configure subscription spec config](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/commit/39a020d528bfdd7ac86439f52f83df01818e18fd)
- [allow to configure subscription starting csv](https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/commit/016b3a9da6ff33b80f4df8d0e033da5fedf65f2e)

This follows up https://github.com/bf2fc6cc711aee1a0c2a/kas-installer/pull/161/files/089afee335a31701ecd840b9ec94f689695f3af3#r859214377

The strimzi operator subscription will now get installed and it'll look like:
```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  annotations:
    kas-fleet-manager/last-applied-resource-configuration: >-
      {"kind":"Subscription","apiVersion":"operators.coreos.com/v1alpha1","metadata":{"name":"managed-kafka-sub","namespace":"redhat-managed-kafka-operator","creationTimestamp":null},"spec":{"source":"managed-kafka-cs","sourceNamespace":"openshift-marketplace","name":"managed-kafka","channel":"alpha","installPlanApproval":"Automatic","config":{"resources":{"limits":{"cpu":"500m","memory":"1Gi"},"requests":{"cpu":"500m","memory":"1Gi"}}}},"status":{"lastUpdated":null}}
  resourceVersion: '2900325'
  name: managed-kafka-sub
  uid: e08e6f7d-61d8-4cc7-a1a4-66acc0220699
  creationTimestamp: '2022-04-27T12:46:40Z'
  generation: 1
  managedFields:
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:kas-fleet-manager/last-applied-resource-configuration': {}
        'f:spec':
          .: {}
          'f:channel': {}
          'f:config':
            .: {}
            'f:resources':
              .: {}
              'f:limits':
                .: {}
                'f:cpu': {}
                'f:memory': {}
              'f:requests':
                .: {}
                'f:cpu': {}
                'f:memory': {}
          'f:installPlanApproval': {}
          'f:name': {}
          'f:source': {}
          'f:sourceNamespace': {}
      manager: kas-fleet-manager
      operation: Update
      time: '2022-04-27T12:46:40Z'
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:labels':
            .: {}
            'f:operators.coreos.com/managed-kafka.redhat-managed-kafka-operator': {}
      manager: olm
      operation: Update
      time: '2022-04-27T12:46:40Z'
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:status':
          'f:installedCSV': {}
          'f:currentCSV': {}
          'f:catalogHealth': {}
          'f:installPlanRef':
            .: {}
            'f:apiVersion': {}
            'f:kind': {}
            'f:name': {}
            'f:namespace': {}
            'f:resourceVersion': {}
            'f:uid': {}
          'f:installPlanGeneration': {}
          'f:conditions': {}
          .: {}
          'f:installplan':
            .: {}
            'f:apiVersion': {}
            'f:kind': {}
            'f:name': {}
            'f:uuid': {}
          'f:lastUpdated': {}
          'f:state': {}
      manager: catalog
      operation: Update
      subresource: status
      time: '2022-04-27T12:47:54Z'
  namespace: redhat-managed-kafka-operator
  labels:
    operators.coreos.com/managed-kafka.redhat-managed-kafka-operator: ''
spec:
  channel: alpha
  config:
    resources:
      limits:
        cpu: 500m
        memory: 1Gi
      requests:
        cpu: 500m
        memory: 1Gi
  installPlanApproval: Automatic
  name: managed-kafka
  source: managed-kafka-cs
  sourceNamespace: openshift-marketplace
```

and the kas-fleetshard one:
```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  annotations:
    kas-fleet-manager/last-applied-resource-configuration: >-
      {"kind":"Subscription","apiVersion":"operators.coreos.com/v1alpha1","metadata":{"name":"kas-fleetshard-operator-sub","namespace":"redhat-kas-fleetshard-operator","creationTimestamp":null},"spec":{"source":"kas-fleetshard-operator-cs","sourceNamespace":"openshift-marketplace","name":"kas-fleetshard-operator","channel":"alpha","installPlanApproval":"Automatic","config":{"resources":{"limits":{"cpu":"500m","memory":"1Gi"},"requests":{"cpu":"500m","memory":"1Gi"}},"env":[{"name":"SSO_ENABLED","value":"true"},{"name":"MANAGEDKAFKA_ADMINSERVER_EDGE_TLS_ENABLED","value":"true"},{"name":"STANDARD_KAFKA_CONTAINER_CPU","value":"500m"},{"name":"STANDARD_KAFKA_CONTAINER_MEMORY","value":"4Gi"},{"name":"STANDARD_KAFKA_JVM_XMS","value":"1G"},{"name":"STANDARD_ZOOKEEPER_CONTAINER_CPU","value":"500m"}]}},"status":{"lastUpdated":null}}
  resourceVersion: '2900493'
  name: kas-fleetshard-operator-sub
  uid: f09e2aef-0295-4ca1-a979-81a5a96c72a6
  creationTimestamp: '2022-04-27T12:46:40Z'
  generation: 1
  managedFields:
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:kas-fleet-manager/last-applied-resource-configuration': {}
        'f:spec':
          .: {}
          'f:channel': {}
          'f:config':
            .: {}
            'f:env': {}
            'f:resources':
              .: {}
              'f:limits':
                .: {}
                'f:cpu': {}
                'f:memory': {}
              'f:requests':
                .: {}
                'f:cpu': {}
                'f:memory': {}
          'f:installPlanApproval': {}
          'f:name': {}
          'f:source': {}
          'f:sourceNamespace': {}
      manager: kas-fleet-manager
      operation: Update
      time: '2022-04-27T12:46:40Z'
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:labels':
            .: {}
            'f:operators.coreos.com/kas-fleetshard-operator.redhat-kas-fleetshard-operator': {}
      manager: olm
      operation: Update
      time: '2022-04-27T12:46:40Z'
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:status':
          'f:installedCSV': {}
          'f:currentCSV': {}
          'f:catalogHealth': {}
          'f:installPlanRef':
            .: {}
            'f:apiVersion': {}
            'f:kind': {}
            'f:name': {}
            'f:namespace': {}
            'f:resourceVersion': {}
            'f:uid': {}
          'f:installPlanGeneration': {}
          'f:conditions': {}
          .: {}
          'f:installplan':
            .: {}
            'f:apiVersion': {}
            'f:kind': {}
            'f:name': {}
            'f:uuid': {}
          'f:lastUpdated': {}
          'f:state': {}
      manager: catalog
      operation: Update
      subresource: status
      time: '2022-04-27T12:47:51Z'
  namespace: redhat-kas-fleetshard-operator
  labels:
    operators.coreos.com/kas-fleetshard-operator.redhat-kas-fleetshard-operator: ''
spec:
  channel: alpha
  config:
    env:
      - name: SSO_ENABLED
        value: 'true'
      - name: MANAGEDKAFKA_ADMINSERVER_EDGE_TLS_ENABLED
        value: 'true'
      - name: STANDARD_KAFKA_CONTAINER_CPU
        value: 500m
      - name: STANDARD_KAFKA_CONTAINER_MEMORY
        value: 4Gi
      - name: STANDARD_KAFKA_JVM_XMS
        value: 1G
      - name: STANDARD_ZOOKEEPER_CONTAINER_CPU
        value: 500m
    resources:
      limits:
        cpu: 500m
        memory: 1Gi
      requests:
        cpu: 500m
        memory: 1Gi
  installPlanApproval: Automatic
  name: kas-fleetshard-operator
  source: kas-fleetshard-operator-cs
  sourceNamespace: openshift-marketplace

```

notice the `.spec.config` part.

/cc @MikeEdgar can you have a look? 

## Verification Steps

A verification from @MikeEdgar will be enough as far as the integration with kas-installer is concerned

## Checklist (Definition of Done)

- ~~[ ] All acceptance criteria specified in JIRA have been completed~~
- ~~[ ] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)~~
- [x] Documentation added for the feature
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer
- ~~[ ] Required metrics/dashboards/alerts have been added (or PR created).~~
- ~~[ ] Required Standard Operating Procedure (SOP) is added.~~
- ~~[ ] JIRA has created for changes required on the client side~~

### CODE REVIEW

File: internal/kafka/internal/config/dataplane_cluster_config.go [hunk 1]
```diff
@@ -366,6 +432,19 @@ func readDataPlaneClusterConfig(file string) (ClusterList, error) {
 	}
 }
 
+func readOperatorsSubscriptionConfigFile(file string, subscriptionConfig *operatorsv1alpha1.SubscriptionConfig) error {
+	fileContents, err := shared.ReadFile(file)
+	if err != nil {
+		return err
+	}
+
+	if err = k8sYaml.UnmarshalStrict([]byte(fileContents), &subscriptionConfig); err != nil {
```
Comment (Rajagopalan-Ranganathan) [inline internal/kafka/internal/config/dataplane_cluster_config.go:441]: can we just return `return k8sYaml.UnmarshalStrict([]byte(fileContents), &subscriptionConfig)` that would return an error or nil

Comment (machi1990) [reply internal/kafka/internal/config/dataplane_cluster_config.go:441]: You are totally right. Thanks

Comment (machi1990) [reply internal/kafka/internal/config/dataplane_cluster_config.go:441]: done

**Comment by machi1990** [comment]
FYI @daveconde this could be of interest to you as well once merged. The reasoning for it are explained in https://github.com/bf2fc6cc711aee1a0c2a/kas-installer/pull/161#discussion_r859214377

File: internal/kafka/internal/config/dataplane_cluster_config.go [hunk 2]
```diff
@@ -85,13 +95,47 @@ func NewDataplaneClusterConfig() *DataplaneClusterConfig {
...
+						v1.ResourceMemory: resource.MustParse("1Gi"),
+						v1.ResourceCPU:    resource.MustParse("500m"),
+					},
+					Limits: v1.ResourceList{
+						v1.ResourceMemory: resource.MustParse("1Gi"),
+						v1.ResourceCPU:    resource.MustParse("500m"),
+					},
+				},
+			},
+			SubscriptionConfigFile: "config/strimzi-operator-subscription-spec-config.yaml",
```
Comment (MikeEdgar) [inline internal/kafka/internal/config/dataplane_cluster_config.go:110]: I'm not so sure about specifying default values (in code or in YAML). These resources requests/limits and the env variables from the YAML would be applied if nothing is specified by the user's configuration, correct?

Comment (machi1990) [reply internal/kafka/internal/config/dataplane_cluster_config.go:110]: Yes. If nothing is specified by the user, the default resources limits e.g in https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pull/971/files#diff-fef0e0515f4cddc045b1e59b5302c05fb909a1c85e5aed861b07b2782928a206R98 will be used.

Comment (machi1990) [reply internal/kafka/internal/config/dataplane_cluster_config.go:110]: > These resources requests/limits and the env variables from the YAML would be applied if nothing is specified by the user's configuration

The user is to supply/configure this via the YAML file. For local development, our config files lives in `config/` folder. 
For when deploying in OpenShift (the case for kas-installer), the value from the ConfigMap, created via the exposed parameters will be used.

When the YAML is empty or does not exist, the default values defined in: 
- https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/195ed09316951c89e6253cb9e3fb80a0c5e489d3/internal/kafka/internal/config/dataplane_cluster_config.go#L98-L109 and
- https://github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/blob/195ed09316951c89e6253cb9e3fb80a0c5e489d3/internal/kafka/internal/config/dataplane_cluster_config.go#L119-L139

will be used.

Comment (machi1990) [reply internal/kafka/internal/config/dataplane_cluster_config.go:110]: I'll go ahead and remove the defaults from the code and let everything be configured via the yaml.

However, I'll leave the value defined in `config/*.yaml` file for local development.

File: templates/service-template.yml [hunk 3]
```diff
@@ -539,6 +539,24 @@ parameters:
...
 
+- name: STRIMZI_OPERATOR_SUBSCRIPTION_CONFIG
+  displayName: Strimzi operator subscription config
+  description: Strimzi operator subscription config. This is applied for standalone clusters only. The configuration must be of type https://pkg.go.dev/github.com/operator-framework/api@v0.3.25/pkg/operators/v1alpha1?utm_source=gopls#SubscriptionConfig
+  value: "{}"
+
+- name: KAS_FLEETSHARD_OPERATOR_SUBSCRIPTION_CONFIG
+  displayName: Kas-fleetshard operator subscription config
+  description: kas-fleetshard operator subscription config. This is applied for standalone clusters only. The configuration must be of type https://pkg.go.dev/github.com/operator-framework/api@v0.3.25/pkg/operators/v1alpha1?utm_source=gopls#SubscriptionConfig
+  value: "{}"
```
Comment (machi1990) [inline templates/service-template.yml:550]: @MikeEdgar the 
- `STRIMZI_OPERATOR_SUBSCRIPTION_CONFIG` and
- `KAS_FLEETSHARD_OPERATOR_SUBSCRIPTION_CONFIG`

parameters are what can be used to configure the subscription spec config for the two operators.

Comment (MikeEdgar) [reply templates/service-template.yml:550]: Thank you - I'll pull the latest and try it out now

Comment (machi1990) [reply templates/service-template.yml:550]: Thank you.

**Comment by machi1990** [comment]
@MikeEdgar the last commit changes the catalog source namespace and uses the operators namespace.
Here is what I've against a crc cluster
```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  annotations:
    kas-fleet-manager/last-applied-resource-configuration: >-
      {"kind":"CatalogSource","apiVersion":"operators.coreos.com/v1alpha1","metadata":{"name":"kas-fleetshard-operator-cs","namespace":"redhat-kas-fleetshard-operator","creationTimestamp":null},"spec":{"sourceType":"grpc","image":"quay.io/osd-addons/kas-fleetshard-operator:production-82b42db","icon":{"base64data":"","mediatype":""}},"status":{}}
  creationTimestamp: '2022-04-27T15:33:27Z'
  generation: 1
  managedFields:
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:kas-fleet-manager/last-applied-resource-configuration': {}
        'f:spec':
          .: {}
          'f:icon':
            .: {}
            'f:base64data': {}
            'f:mediatype': {}
          'f:image': {}
          'f:sourceType': {}
      manager: kas-fleet-manager
      operation: Update
      time: '2022-04-27T15:33:27Z'
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:status':
          .: {}
          'f:connectionState':
            .: {}
            'f:address': {}
            'f:lastConnect': {}
            'f:lastObservedState': {}
          'f:registryService':
            .: {}
            'f:createdAt': {}
            'f:port': {}
            'f:protocol': {}
            'f:serviceName': {}
            'f:serviceNamespace': {}
      manager: catalog
      operation: Update
      subresource: status
      time: '2022-04-27T15:33:28Z'
  name: kas-fleetshard-operator-cs
  namespace: redhat-kas-fleetshard-operator
  resourceVersion: '45309'
  uid: 79f157a7-baf0-4b7c-83df-db4f519b6f5f
spec:
  icon:
    base64data: ''
    mediatype: ''
  image: 'quay.io/osd-addons/kas-fleetshard-operator:production-82b42db'
  sourceType: grpc
status:
  connectionState:
    address: 'kas-fleetshard-operator-cs.redhat-kas-fleetshard-operator.svc:50051'
    lastConnect: '2022-04-27T15:46:16Z'
    lastObservedState: READY
  registryService:
    createdAt: '2022-04-27T15:33:27Z'
    port: '50051'
    protocol: grpc
    serviceName: kas-fleetshard-operator-cs
    serviceNamespace: redhat-kas-fleetshard-operator
```

```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  annotations:
    kas-fleet-manager/last-applied-resource-configuration: >-
      {"kind":"Subscription","apiVersion":"operators.coreos.com/v1alpha1","metadata":{"name":"kas-fleetshard-operator-sub","namespace":"redhat-kas-fleetshard-operator","creationTimestamp":null},"spec":{"source":"kas-fleetshard-operator-cs","sourceNamespace":"redhat-kas-fleetshard-operator","name":"kas-fleetshard-operator","channel":"alpha","installPlanApproval":"Automatic","config":{"resources":{"limits":{"cpu":"500m","memory":"1Gi"},"requests":{"cpu":"500m","memory":"1Gi"}},"env":[{"name":"SSO_ENABLED","value":"true"},{"name":"MANAGEDKAFKA_ADMINSERVER_EDGE_TLS_ENABLED","value":"true"},{"name":"STANDARD_KAFKA_CONTAINER_CPU","value":"500m"},{"name":"STANDARD_KAFKA_CONTAINER_MEMORY","value":"4Gi"},{"name":"STANDARD_KAFKA_JVM_XMS","value":"1G"},{"name":"STANDARD_ZOOKEEPER_CONTAINER_CPU","value":"500m"}]}},"status":{"lastUpdated":null}}
  resourceVersion: '40645'
  name: kas-fleetshard-operator-sub
  uid: 61e47a6a-b679-4002-ac92-dae785e1a1d7
  creationTimestamp: '2022-04-27T15:33:27Z'
  generation: 1
  managedFields:
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:kas-fleet-manager/last-applied-resource-configuration': {}
        'f:spec':
          .: {}
          'f:channel': {}
          'f:config':
            .: {}
            'f:env': {}
            'f:resources':
              .: {}
              'f:limits':
                .: {}
                'f:cpu': {}
                'f:memory': {}
              'f:requests':
                .: {}
                'f:cpu': {}
                'f:memory': {}
          'f:installPlanApproval': {}
          'f:name': {}
          'f:source': {}
          'f:sourceNamespace': {}
      manager: kas-fleet-manager
      operation: Update
      time: '2022-04-27T15:33:27Z'
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:labels':
            .: {}
            'f:operators.coreos.com/kas-fleetshard-operator.redhat-kas-fleetshard-operator': {}
      manager: olm
      operation: Update
      time: '2022-04-27T15:33:27Z'
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:status':
          'f:installedCSV': {}
          'f:currentCSV': {}
          'f:catalogHealth': {}
          'f:installPlanRef':
            .: {}
            'f:apiVersion': {}
            'f:kind': {}
            'f:name': {}
            'f:namespace': {}
            'f:resourceVersion': {}
            'f:uid': {}
          'f:installPlanGeneration': {}
          'f:conditions': {}
          .: {}
          'f:installplan':
            .: {}
            'f:apiVersion': {}
            'f:kind': {}
            'f:name': {}
            'f:uuid': {}
          'f:lastUpdated': {}
          'f:state': {}
      manager: catalog
      operation: Update
      subresource: status
      time: '2022-04-27T15:34:25Z'
  namespace: redhat-kas-fleetshard-operator
  labels:
    operators.coreos.com/kas-fleetshard-operator.redhat-kas-fleetshard-operator: ''
spec:
  channel: alpha
  config:
    env:
      - name: SSO_ENABLED
        value: 'true'
      - name: MANAGEDKAFKA_ADMINSERVER_EDGE_TLS_ENABLED
        value: 'true'
      - name: STANDARD_KAFKA_CONTAINER_CPU
        value: 500m
      - name: STANDARD_KAFKA_CONTAINER_MEMORY
        value: 4Gi
      - name: STANDARD_KAFKA_JVM_XMS
        value: 1G
      - name: STANDARD_ZOOKEEPER_CONTAINER_CPU
        value: 500m
    resources:
      limits:
        cpu: 500m
        memory: 1Gi
      requests:
        cpu: 500m
        memory: 1Gi
  installPlanApproval: Automatic
  name: kas-fleetshard-operator
  source: kas-fleetshard-operator-cs
  sourceNamespace: redhat-kas-fleetshard-operator
status:
  installplan:
    apiVersion: operators.coreos.com/v1alpha1
    kind: InstallPlan
    name: install-jkmvg
    uuid: e8fa7819-7c9b-4e1e-adbc-4fab576dc0e4
  lastUpdated: '2022-04-27T15:34:28Z'
  installedCSV: kas-fleetshard-operator.0.7.0-2
  currentCSV: kas-fleetshard-operator.0.7.0-2
  installPlanRef:
    apiVersion: operators.coreos.com/v1alpha1
    kind: InstallPlan
    name: install-jkmvg
    namespace: redhat-kas-fleetshard-operator
    resourceVersion: '40394'
    uid: e8fa7819-7c9b-4e1e-adbc-4fab576dc0e4
  state: AtLatestKnown
  catalogHealth:
    - catalogSourceRef:
        apiVersion: operators.coreos.com/v1alpha1
        kind: CatalogSource
        name: certified-operators
        namespace: openshift-marketplace
        resourceVersion: '38779'
        uid: bb49c49e-54ae-479f-a011-dbfebb058c56
      healthy: true
      lastUpdated: '2022-04-27T15:33:27Z'
    - catalogSourceRef:
        apiVersion: operators.coreos.com/v1alpha1
        kind: CatalogSource
        name: community-operators
        namespace: openshift-marketplace
        resourceVersion: '38572'
        uid: f028cfde-68c3-4108-af91-9c7a4323e61d
      healthy: true
      lastUpdated: '2022-04-27T15:33:27Z'
    - catalogSourceRef:
        apiVersion: operators.coreos.com/v1alpha1
        kind: CatalogSource
        name: redhat-marketplace
        namespace: openshift-marketplace
        resourceVersion: '39535'
        uid: e1f0786b-bd64-4be6-a725-6c6deb720f55
      healthy: true
      lastUpdated: '2022-04-27T15:33:27Z'
    - catalogSourceRef:
        apiVersion: operators.coreos.com/v1alpha1
        kind: CatalogSource
        name: redhat-operators
        namespace: openshift-marketplace
        resourceVersion: '38674'
        uid: d44889c4-48f5-4af4-ace2-52a7b08ecd0e
      healthy: true
      lastUpdated: '2022-04-27T15:33:27Z'
    - catalogSourceRef:
        apiVersion: operators.coreos.com/v1alpha1
        kind: CatalogSource
        name: kas-fleetshard-operator-cs
        namespace: redhat-kas-fleetshard-operator
        resourceVersion: '40034'
        uid: 79f157a7-baf0-4b7c-83df-db4f519b6f5f
      healthy: true
      lastUpdated: '2022-04-27T15:33:27Z'
  conditions:
    - lastTransitionTime: '2022-04-27T15:33:27Z'
      message: all available catalogsources are healthy
      reason: AllCatalogSourcesHealthy
      status: 'False'
      type: CatalogSourcesUnhealthy
  installPlanGeneration: 1
```

And for strimzi
```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: CatalogSource
metadata:
  annotations:
    kas-fleet-manager/last-applied-resource-configuration: >-
      {"kind":"CatalogSource","apiVersion":"operators.coreos.com/v1alpha1","metadata":{"name":"managed-kafka-cs","namespace":"redhat-managed-kafka-operator","creationTimestamp":null},"spec":{"sourceType":"grpc","image":"quay.io/osd-addons/managed-kafka:production-82b42db","icon":{"base64data":"","mediatype":""}},"status":{}}
  creationTimestamp: '2022-04-27T15:54:13Z'
  generation: 1
  managedFields:
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:kas-fleet-manager/last-applied-resource-configuration': {}
        'f:spec':
          .: {}
          'f:icon':
            .: {}
            'f:base64data': {}
            'f:mediatype': {}
          'f:image': {}
          'f:sourceType': {}
      manager: kas-fleet-manager
      operation: Update
      time: '2022-04-27T15:54:13Z'
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:status':
          .: {}
          'f:connectionState':
            .: {}
            'f:address': {}
            'f:lastConnect': {}
            'f:lastObservedState': {}
          'f:registryService':
            .: {}
            'f:createdAt': {}
            'f:port': {}
            'f:protocol': {}
            'f:serviceName': {}
            'f:serviceNamespace': {}
      manager: catalog
      operation: Update
      subresource: status
      time: '2022-04-27T15:54:14Z'
  name: managed-kafka-cs
  namespace: redhat-managed-kafka-operator
  resourceVersion: '47475'
  uid: e7c26779-62bc-47de-bd06-b2c4318122c6
spec:
  icon:
    base64data: ''
    mediatype: ''
  image: 'quay.io/osd-addons/managed-kafka:production-82b42db'
  sourceType: grpc
status:
  connectionState:
    address: 'managed-kafka-cs.redhat-managed-kafka-operator.svc:50051'
    lastConnect: '2022-04-27T15:54:25Z'
    lastObservedState: TRANSIENT_FAILURE
  registryService:
    createdAt: '2022-04-27T15:54:13Z'
    port: '50051'
    protocol: grpc
    serviceName: managed-kafka-cs
    serviceNamespace: redhat-managed-kafka-operator
```

```yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  annotations:
    kas-fleet-manager/last-applied-resource-configuration: >-
      {"kind":"Subscription","apiVersion":"operators.coreos.com/v1alpha1","metadata":{"name":"managed-kafka-sub","namespace":"redhat-managed-kafka-operator","creationTimestamp":null},"spec":{"source":"managed-kafka-cs","sourceNamespace":"redhat-managed-kafka-operator","name":"managed-kafka","channel":"alpha","installPlanApproval":"Automatic","config":{"resources":{"limits":{"cpu":"500m","memory":"1Gi"},"requests":{"cpu":"500m","memory":"1Gi"}}}},"status":{"lastUpdated":null}}
  resourceVersion: '47665'
  name: managed-kafka-sub
  uid: 50138050-73aa-4ea2-817f-2f1dd1c875cb
  creationTimestamp: '2022-04-27T15:54:13Z'
  generation: 1
  managedFields:
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:annotations':
            .: {}
            'f:kas-fleet-manager/last-applied-resource-configuration': {}
        'f:spec':
          .: {}
          'f:channel': {}
          'f:config':
            .: {}
            'f:resources':
              .: {}
              'f:limits':
                .: {}
                'f:cpu': {}
                'f:memory': {}
              'f:requests':
                .: {}
                'f:cpu': {}
                'f:memory': {}
          'f:installPlanApproval': {}
          'f:name': {}
          'f:source': {}
          'f:sourceNamespace': {}
      manager: kas-fleet-manager
      operation: Update
      time: '2022-04-27T15:54:13Z'
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:metadata':
          'f:labels':
            .: {}
            'f:operators.coreos.com/managed-kafka.redhat-managed-kafka-operator': {}
      manager: olm
      operation: Update
      time: '2022-04-27T15:54:13Z'
    - apiVersion: operators.coreos.com/v1alpha1
      fieldsType: FieldsV1
      fieldsV1:
        'f:status':
          'f:installedCSV': {}
          'f:currentCSV': {}
          'f:catalogHealth': {}
          'f:installPlanRef':
            .: {}
            'f:apiVersion': {}
            'f:kind': {}
            'f:name': {}
            'f:namespace': {}
            'f:resourceVersion': {}
            'f:uid': {}
          'f:installPlanGeneration': {}
          'f:conditions': {}
          .: {}
          'f:installplan':
            .: {}
            'f:apiVersion': {}
            'f:kind': {}
            'f:name': {}
            'f:uuid': {}
          'f:lastUpdated': {}
          'f:state': {}
      manager: catalog
      operation: Update
      subresource: status
      time: '2022-04-27T15:54:41Z'
  namespace: redhat-managed-kafka-operator
  labels:
    operators.coreos.com/managed-kafka.redhat-managed-kafka-operator: ''
spec:
  channel: alpha
  config:
    resources:
      limits:
        cpu: 500m
        memory: 1Gi
      requests:
        cpu: 500m
        memory: 1Gi
  installPlanApproval: Automatic
  name: managed-kafka
  source: managed-kafka-cs
  sourceNamespace: redhat-managed-kafka-operator
status:
  installplan:
    apiVersion: operators.coreos.com/v1alpha1
    kind: InstallPlan
    name: install-pthdz
    uuid: c00069b8-0363-4685-990d-0db9503fa56e
  lastUpdated: '2022-04-27T15:54:41Z'
  installedCSV: strimzi-cluster-operator.v0.22.1-6
  currentCSV: strimzi-cluster-operator.v0.22.1-6
  installPlanRef:
    apiVersion: operators.coreos.com/v1alpha1
    kind: InstallPlan
    name: install-pthdz
    namespace: redhat-managed-kafka-operator
    resourceVersion: '47557'
    uid: c00069b8-0363-4685-990d-0db9503fa56e
  state: AtLatestKnown
  catalogHealth:
    - catalogSourceRef:
        apiVersion: operators.coreos.com/v1alpha1
        kind: CatalogSource
        name: certified-operators
        namespace: openshift-marketplace
        resourceVersion: '45736'
        uid: bb49c49e-54ae-479f-a011-dbfebb058c56
      healthy: true
      lastUpdated: '2022-04-27T15:54:16Z'
    - catalogSourceRef:
        apiVersion: operators.coreos.com/v1alpha1
        kind: CatalogSource
        name: community-operators
        namespace: openshift-marketplace
        resourceVersion: '45738'
        uid: f028cfde-68c3-4108-af91-9c7a4323e61d
      healthy: true
      lastUpdated: '2022-04-27T15:54:16Z'
    - catalogSourceRef:
        apiVersion: operators.coreos.com/v1alpha1
        kind: CatalogSource
        name: redhat-marketplace
        namespace: openshift-marketplace
        resourceVersion: '46850'
        uid: e1f0786b-bd64-4be6-a725-6c6deb720f55
      healthy: true
      lastUpdated: '2022-04-27T15:54:16Z'
    - catalogSourceRef:
        apiVersion: operators.coreos.com/v1alpha1
        kind: CatalogSource
        name: redhat-operators
        namespace: openshift-marketplace
        resourceVersion: '45978'
        uid: d44889c4-48f5-4af4-ace2-52a7b08ecd0e
      healthy: true
      lastUpdated: '2022-04-27T15:54:16Z'
    - catalogSourceRef:
        apiVersion: operators.coreos.com/v1alpha1
        kind: CatalogSource
        name: managed-kafka-cs
        namespace: redhat-managed-kafka-operator
        resourceVersion: '47400'
        uid: e7c26779-62bc-47de-bd06-b2c4318122c6
      healthy: true
      lastUpdated: '2022-04-27T15:54:16Z'
  conditions:
    - lastTransitionTime: '2022-04-27T15:54:16Z'
      message: all available catalogsources are healthy
      reason: AllCatalogSourcesHealthy
      status: 'False'
      type: CatalogSourcesUnhealthy
    - lastTransitionTime: '2022-04-27T15:54:39Z'
      reason: Installing
      status: 'True'
      type: InstallPlanPending
  installPlanGeneration: 1

```

**Review by MikeEdgar — APPROVED** [review]
This is working well for me, thank you :heavy_check_mark:

**Pull request merged**